server/embed/config_test.go

```diff
@@ -559,14 +560,40 @@ func TestInferLocalAddr(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			cfg := NewConfig()
 			cfg.AdvertisePeerUrls = types.MustNewURLs(tt.advertisePeerURLs)
-			cfg.ExperimentalSetMemberLocalAddr = tt.setMemberLocalAddr
+			cfg.ServerFeatureGate.(featuregate.MutableFeatureGate).Set(tt.serverFeatureGates)
 
 			require.NoError(t, cfg.Validate())
 			require.Equal(t, tt.expectedLocalAddr, cfg.InferLocalAddr())
 		})
 	}
 }
 
+func TestSetMemberLocalAddrValidate(t *testing.T) {
+	tcs := []struct {
+		name               string
+		serverFeatureGates string
+		expectError        bool
```

Comment:
You add this field, but it isn't set at all, which means it's aways `false`. Suggest to just remove it.